fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -147,17 +147,6 @@ def parse_line(line, hide_output)
         end
       end
     end
-
-    def additional_upload_parameters
```

Comment:
Hey @mollyIV 

I think backwards compatibility is resolved by removing the older workaround, so that by default no `-t` transport option is added, but anyone using that environment variable to add their own will actually have it honoured (which was the case in my instance)

As far as the iTMSTransporter/Xcode version, iTMSTransporter 2.0 was released on the 16th, but our builds only started failing to upload to TestFlight with this issue today, and based on the revision history released by Apple, it appears to be an undocumented change that's breaking.

However, in relation to automatically detemining what transport to use, that looks like it was added to iTMSTransporter 1.5.0 in 2009 (!) - see https://help.apple.com/itc/transporteruserguide/#/apdA1aded1d2 near the bottom of the page.

Reply:
@mollyIV I am responding - since I also did the same change - but dropped my PR in favour of @silvervest 


https://help.apple.com/itc/transporteruserguide/en.lproj/static.html#apdA1aded1d2

The Transporter Guide for 2.0 includes the following changes:

HTTP transport is now supported and server IPs are listed in System and network requirements. HTTP transport is faster than the DAV transport and doesn't require firewall re-configuration.

OpenJDK 11 is now included for Linux and Windows, in addition to macOS. Linux and Windows users no longer need to download and install a separate JDK to run Transporter. Starting with this release, Transporter can only be installed on 64-bit (x86-64 architecture) operating systems.

Server IPs added. New server IPs have been added for Aspera and Signiant. See System and network requirements.

Note: Previously installed versions of Transporter that use JDK 8, 9, or 10 will continue to auto update to Transporter 2.0. Apple strongly recommends that you update your operating system and Transporter to newer releases so that you will continue to receive automatic updates with performance and security fixes.

Reply:
> However, in relation to automatically detemining what transport to use, that looks like it was added to iTMSTransporter 1.5.0 in 2009 (!) - see [help.apple.com/itc/transporteruserguide/#/apdA1aded1d2](https://help.apple.com/itc/transporteruserguide/#/apdA1aded1d2) near the bottom of the page.

If I remember correctly fastlane chose to override that because the automatic detection was wonky (or even unusable for some) in the past - which explains the existing behavior.

Reply:
@janpio - True, but on 1 march 2018  there was some hiccups with relates to the same issue and then this workaround was added .

Reply:
Having said that, I am now thinking of the backwards compatibility issue of fastlane previously defaulting to using Signiant as the only transport method instead of letting iTMSTransporter decide what to do.

If that's preferred, I could amend the change to instead process the `DELIVER_ITMSTRANSPORTER_ADDITIONAL_UPLOAD_PARAMETERS` environment variable and glue any additional `-t` parameters together in a comma-delimited list and pass them along with Signiant added to the end as the fallback?
`DELIVER_ITMSTRANSPORTER_ADDITIONAL_UPLOAD_PARAMETERS=-t DAV` would become `-t DAV,Signiant` for example.

Reply:
When i tested it in my local machine, it is switching from one transport option to another to find the one which works properly and then uses it.

Reply:
My suggestion is not to add any transport option from fastlane (exception could be the env variable)- because all transporter are getting auto updated. 

But will leave the option to @mollyIV and @janpio - as they are better judges in this.

Reply:
Alternatively, we could put the DAV workaround back in place and simply remove the Signiant default. This would ensure compatibility with the majority of existing use cases, where the environment variable isn't being set, but still allow users to override.

Reply:
What we should definitely achieve is to surface the option to change this much better in the docs - currently this is not _really_ that well documented anywhere I guess.

I don't have an opinion on the other things yet and will have to spend some time figuring things out before I can say more.